internal/pkg/storage/store.go

```diff
@@ -82,3 +91,52 @@ func GetCache(key string) (string, error) {
 	}
 	return string(s), nil
 }
+
+// GetCacheIfWithinTTL gets an item from the cache if it's with the TTL duration. To simplify the TTL is provided by the caller
+// the data store just tracks the key and it's last updated value
+func GetCacheWithTTL(key string, ttl time.Duration) (valid bool, value string, err error) {
+	cacheItem, err := GetCache(key)
+	if err != nil {
+		return false, "", err
+	}
+
+	// Empty string isn't a valid cache item
+	if cacheItem == "" {
+		return false, "", err
+	}
+
+	// Get the Last updated time for this cache key
+	cacheItemLastUpdated, err := GetCache(key + ttlLastUpdatedKey)
+	if err != nil || cacheItemLastUpdated == "" {
+		return false, cacheItem, err
+	}
+	lastUpdatedEpoc, err := strconv.ParseInt(cacheItemLastUpdated, 10, 64)
+	if err != nil {
+		return false, cacheItem, errors.Wrapf(err, "Failed to parse %v", cacheItemLastUpdated)
+	}
+
+	// Check if the ttl has expired
+	ttlExpiresAfter := time.Unix(lastUpdatedEpoc, 0).Add(ttl)
+	if clock.Now().After(ttlExpiresAfter) {
+		// It has
+		return false, cacheItem, nil
+	}
+
+	return true, cacheItem, nil
+}
+
+// PutCacheItemForTTL puts an item in the cache bucket
+func PutCacheForTTL(key, value string) error {
+	// Save the Item
+	err := PutCache(key, value)
+	// Track when it was saved
+	errUpdate := PutCache(key+ttlLastUpdatedKey, epocToString())
```

Comment:
Does anything ever clear out values with expired TTLs?

Reply:
Nope, I initially looked at writing something to do some clearing but felt unnecessary with the current use case as they're the same keys overwritten ever time the app runs. 

The fact that this comes up tho and the naming of the methods means I should put a warning about this in the code as future users might assume that it does. Sound like a good middle ground?

Reply:
Note added to function description